saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/web/Saml2AuthenticationTokenConverter.java

```diff
@@ -50,6 +54,10 @@
 
 	private final Converter<HttpServletRequest, RelyingPartyRegistration> relyingPartyRegistrationResolver;
 
+	private Saml2AuthenticationRequestRepository<AbstractSaml2AuthenticationRequest> authenticationRequestRepository = new HttpSessionSaml2AuthenticationRequestRepository();
```

Comment:
It would be better to remove this reference since it would otherwise need to stay in sync with the `loader` member.

For example, in your constructor you could do:

```java
this.loader = new HttpSessionSaml2AuthenticationRequestFactory()::loadAuthenticationRequest;
```